.github/workflows/lint-test-build.yml

```diff
@@ -0,0 +1,128 @@
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm ci
+        if: steps.cache.outputs.cache-hit != 'true'
+
+      - name: Set up QEMU
+        if: matrix.action == 'build'
+        id: qemu
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: amd64,arm64
+
+      - name: Set up Docker Buildx
+        id: buildx
+        if: matrix.action == 'build'
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+
+      - name: Cache Docker layers
+        id: cache-docker
+        if: steps.buildx.outcome == 'success'
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.temp }}/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.node-version }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ matrix.node-version }}-
+
+      - name: Docker cache size fix
+        if: steps.cache-docker.outcome == 'success'
+        run: |
+          mv -v ${{ runner.temp }}/.buildx-cache ${{ runner.temp }}/.buildx-cache-old || true
+
+      - name: Docker meta
+        if: steps.buildx.outcome == 'success'
+        id: meta
+        uses: docker/metadata-action@v5
+        env:
+          DOCKER_METADATA_SET_OUTPUT_ENV: "true"
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}
+            #${{ github.repository }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=sha
+
+      - name: Set variables for ci:build
+        if: matrix.action == 'build'
+        run: |
+          echo $DOCKER_METADATA_OUTPUT_JSON | jq -r '"DOCKER_TAGS=-t " + (.tags | join(" -t "))' >> $GITHUB_ENV
+          echo $DOCKER_METADATA_OUTPUT_JSON | jq -r '"REVISION=" + (.labels."org.opencontainers.image.revision")' >> $GITHUB_ENV
+          echo $DOCKER_METADATA_OUTPUT_JSON | jq -r '"TAG_VERSION_NUMBER=" + (.labels."org.opencontainers.image.version")' >> $GITHUB_ENV
+          echo $DOCKER_METADATA_OUTPUT_JSON | jq -r '"DATETIMENOW=" + (.labels."org.opencontainers.image.created")' >> $GITHUB_ENV
+          #scope="buildx-${{ matrix.node-version }}-"
+          #echo CACHE_ARGS="--cache-from=type=gha,scope=$scope --cache-to=type=gha,scope=$scope,mode=max" >> $GITHUB_ENV
+          cachesrc="${{ runner.temp }}/.buildx-cache"
+          echo CACHE_ARGS="--cache-from=type=local,src=$cachesrc-old --cache-to=type=local,dest=$cachesrc,mode=max" >> $GITHUB_ENV
+          echo EXTRA_ARGS=--platform linux/amd64,linux/arm64 --push >> $GITHUB_ENV
+
+      # FIXME this is disabled because we only have permission to run 'verified' actions
+      # instead we must use the local cache, which needs to be manually cleared from time to time
+      #- uses: peaceiris/actions-export-envs@v1.1.0
+        #id: envs
```

Comment:
If this is required, why not simply run the [script it does](https://github.com/peaceiris/actions-export-envs/blob/main/entrypoint.sh), i.e.

```
#!/bin/sh

cat << EOF >> "${GITHUB_OUTPUT}"
ACTIONS_RUNTIME_TOKEN=${ACTIONS_RUNTIME_TOKEN}
ACTIONS_CACHE_URL=${ACTIONS_CACHE_URL}
EOF
```

That way we don't need to rely on an action that isn't going through a verification process

Reply:
Could you please try [actions/github-script](https://github.com/actions/github-script) instead, e.g. something like [this](https://stackoverflow.com/a/64652612/664054)?